haystack/document_store/faiss.py

```diff
@@ -29,44 +27,79 @@ def __init__(
                         deployment, Postgres is recommended.
         :param index_buffer_size: When working with large datasets, the ingestion process(FAISS + SQL) can be buffered in
                                   smaller chunks to reduce memory footprint.
-        :param vector_size: the embedding vector size.
-        :param faiss_index: load an existing FAISS Index.
+        :param vector_dim: the embedding vector size.
+        :param faiss_index_factory_str: Create a new FAISS index of the specified type.
+                                        The type is determined from the given string following the conventions
+                                        of the original FAISS index factory.
+                                        Recommended options:
+                                        - "Flat" (default): Best accuracy (= exact). Becomes slow and RAM intense for > 1 Mio docs.
+                                        - "HNSW": Graph-based heuristic. If not further specified,
+                                                  we use a RAM intense, but more accurate config:
+                                                  HNSW256, efConstruction=256 and efSearch=256
+                                        - "IVFx,Flat": Inverted Index. Replace x with the number of centroids aka nlist.
+                                                          Rule of thumb: nlist = 10 * sqrt (num_docs) is a good starting point.
+                                        For more details see:
+                                        - Overview of indices https://github.com/facebookresearch/faiss/wiki/Faiss-indexes
+                                        - Guideline for choosing an index https://github.com/facebookresearch/faiss/wiki/Guidelines-to-choose-an-index
+                                        - FAISS Index factory https://github.com/facebookresearch/faiss/wiki/The-index-factory
+                                        Benchmarks: XXX
+        :param faiss_index: Pass an existing FAISS Index, i.e. an empty one that you configured manually
+                            or one with docs that you used in Haystack before and want to load again.
         """
-        self.vector_size = vector_size
-        self.faiss_index = faiss_index
+        self.vector_dim = vector_dim
+
+        if faiss_index:
+            self.faiss_index = faiss_index
+        else:
+            self.faiss_index = self._create_new_index(vector_dim=self.vector_dim, index_factory=faiss_index_factory_str, **kwargs)
+
         self.index_buffer_size = index_buffer_size
         super().__init__(url=sql_url)
 
-    def _create_new_index(self, vector_size: int, index_factory: str = "HNSW4"):
-        index = faiss.index_factory(vector_size + 1, index_factory)
+    def _create_new_index(self, vector_dim: int, index_factory: str = "Flat", metric=faiss.METRIC_INNER_PRODUCT, **kwargs):
```

Comment:
Better to rename `metric` to `metric_type` , also following doc can be referenced https://github.com/facebookresearch/faiss/wiki/MetricType-and-distances

Reply:
Good point. Thx. Changed it